Quadriga/src/main/java/edu/asu/spring/quadriga/web/config/SecurityContext.java

```diff
@@ -0,0 +1,139 @@
+                    .antMatchers("/auth/workbench/**", "/auth/rest/**", "/auth/conceptcollections/**",
+                            "/auth/conceptdetails/**", "/auth/transformation/**", "/auth/searchitems/**",
+                            "/auth/dictionaries/**", "/auth/editing/**", "/auth/networks/**")
+                    .hasAnyRole("QUADRIGA_USER_STANDARD", "QUADRIGA_USER_COLLABORATOR")
+                    .antMatchers("/users/**", "/admin/**", "/auth/users/**", "/checks/**")
+                    .hasRole("QUADRIGA_USER_ADMIN").anyRequest()
+                    .hasAnyRole("QUADRIGA_USER_ADMIN", "QUADRIGA_USER_STANDARD", "QUADRIGA_USER_COLLABORATOR")
+                    .antMatchers("/**").denyAll()
+                    // Adds the SocialAuthenticationFilter to Spring Security's
+                    // filter chain.
+                    .and().apply(new SpringSocialConfigurer());
+
+        }
+
+        @Autowired
+        @Override
+        protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+            auth.authenticationProvider(daoAuthProvider());
+            auth.eraseCredentials(false);
+            auth.inMemoryAuthentication().withUser("admin").password("admin").roles("QUADRIGA_USER_ADMIN");
+            super.configure(auth);
+
+        }
+
+        @Bean(name = BeanIds.AUTHENTICATION_MANAGER)
+        @Override
+        public AuthenticationManager authenticationManagerBean() throws Exception {
+            return super.authenticationManager();
+        }
+
+        @Bean
+        public PasswordEncoder passwordEncoder() {
+            return new BCryptPasswordEncoder(10);
+        }
+
+        @Bean
+        public SocialUserDetailsService socialUserDetailsService() {
+            return new SimpleSocialUserDetailsService(userDetailsService());
+        }
+
+        @Bean
+        public UserDetailsService userDetailsService() {
+            return userService;
+        }
+
+        @Bean
+        public DaoAuthenticationProvider daoAuthProvider() throws Exception {
+            DaoAuthenticationProvider provider = new DaoAuthenticationProvider();
+            provider.setPasswordEncoder(passwordEncoder());
+            provider.setUserDetailsService(userDetailsService());
+            return provider;
+        }
+
+    }
+
+    // Security Configuration for REST end-points
+    @Configuration
+    @Order(1)
+    public static class StatelessSecurityConfiguration extends WebSecurityConfigurerAdapter {
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            HeadersConfigurer<HttpSecurity> restConfig = http.antMatcher("/rest/**").headers().frameOptions()
+                    .sameOrigin();
+            restConfig.and().sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS);
+            restConfig.and().authorizeRequests().antMatchers("/rest/**").hasAnyRole("QUADRIGA_USER_ADMIN",
```

Comment:
do you really need ```antMatchers("/rest/**")``` again here? you already specify it in line 128.

Reply:
Had there been other URLs starting with /rest ,  it is useful to  explicitly configure /rest/**.
If we are sure every URL starting with /rest needs to be handled the same-way, then we do not need to specify /rest/* in line 131.
I just replicated the XML configurations, which had /rest/** in the http pattern and the intercept-url pattern